### PR TITLE
Add an example for Array.flat()

### PR DESCRIPTION
**Pull request by tigleym**
PR for #1636

### CODE REVIEW

File: live-examples/js-examples/array/array-flat.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const arr1 = [0, 1, 2, [3, 4]];
+
+// Flatten to single level array.
```
Comment (wbamberg) [inline live-examples/js-examples/array/array-flat.js:3]: Generally we don't use comments in interactive examples except for "expected output" or exceptional non-obvious things. I suppose it's a kind of "show don't tell" kind of thing - we have a whole page of docs to say what the feature does, the example is just supposed to show it. So could you remove this line?

File: live-examples/js-examples/array/array-flat.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+console.log(arr1.flat());
+// expected output: [0, 1, 2, 3, 4]
+
+const arr2 = [0, 1, 2, [[[3, 4]]]];
+
+// Flatten to two-level array.
```
Comment (wbamberg) [inline live-examples/js-examples/array/array-flat.js:9]: Same comment for this line.

**Comment by tigleym** [comment]
@wbamberg Thanks for the quick feedback! Please disregard my latest commit since I just submitted it before seeing your feedback.

File: live-examples/js-examples/array/array-flat.js (hunk 1 above)
Comment (tigleym) [reply live-examples/js-examples/array/array-flat.js]: Done!

**Pull request merged**